### ISSUE TITLE
Pollution of global object with connectionIndex property

### ISSUE THREAD

**Issue by rossj** [guidesmiths/rascal#60]
Hi there. I'm working on switching my code to use rascal, and my mocha tests started warning about a global object leak of a "connectionIndex" property.

Adding the `--check-leaks` mocha option to rascal's own tests cause a similar issue:

```
  1) Broker
       should assert vhosts:
     Error: global leak detected: 'connectionIndex'
      at /c/Users/ross/code/rascal/test/broker.tests.js:95:7
      at /c/Users/ross/code/rascal/test/broker.tests.js:250:7
      at Immediate._onImmediate (lib/amqp/Broker.js:1:20202)
      at processImmediate (internal/timers.js:439:21)
```

I believe the culprit is [this line](https://github.com/guidesmiths/rascal/blob/0fdc155c82221007d592e11e5d8ced0fe217f633/lib/amqp/Vhost.js#L55), where `this` ends up being the global object. I'm assuming it should be `self...` instead.

**Comment by cressie176** [comment]
Thanks @rossj. Fixing now

**Comment by cressie176** [comment]
Should be fixed in rascal@4.6.0

**Comment by rossj** [comment]
Looks good. Thanks!
